runtime/tests/checker/member_test.go

```diff
@@ -418,3 +418,299 @@ func TestCheckMemberNotDeclaredSecondaryError(t *testing.T) {
 		assert.Equal(t, "unknown member", memberErr.SecondaryError())
 	})
 }
+
+func TestCheckMemberAccess(t *testing.T) {
```

Comment:
This has test cases for each of {Dictionary, Array, Composite} container types, but only tests access for Array and Dictionary member types. We should have tests for Composite and Restricted type members as well.

Reply:
added in 9a64a7fddffd31976ee9abc25caaaf650083fd68